.github/workflows/ci-build-image.yml

```diff
@@ -64,8 +63,13 @@ jobs:
     - name: maven-settings
       uses: s4u/maven-settings-action@v2
       with:
-        servers: '[{"id": "github", "username": "dummy", "password": "${env.GITHUB_TOKEN_REF}"}]'
-        githubServer: false
+        githubServer: true
+    - name: ghcr login
+      uses: redhat-actions/podman-login@v1
```

Comment:
What causes this to be required now?

Reply:
I thought by setting `githubServer: false`,   the Mvn settings will not include GitHub Packages as a repository, which is needed and that Maven won't  fetch artifacts from GitHub Packages.

Reply:
I'm not sure I understand... where does podman come into that equation?

Reply:
if I have it as `false` on the `integrated-test ` file I get this error: I am checking it again ATM. 
![Screenshot from 2023-10-26 10-03-30](https://github.com/cryostatio/cryostat/assets/56743004/59e4cff7-3e1e-4436-b2dc-c26c842b721f)

Reply:
Also, 

> I'm wondering what happened to this workflow that requires the new step to have podman log in to ghcr.io.

The new step to have podman log in to ghcr.io is to be able to pull the container image. Did I address the question right or may be I din't understand?

Reply:
Right, I'm only asking about the podman part, not about the `githubServer` maven settings part. It makes sense that a step called `build-image` needs to be able to connect to GitHub Packages maven repo in order to pull the -core dependency and build the image.

But I'm not sure I understand why a step called `build-image` requires pulling the same image, and I don't see where that new requirement for this step came from - it doesn't look like the step itself really changed, so I suppose it comes from some other reusable workflow step that this one calls into. If that's the case, should the login step not be moved closer to where it is actually required?